### PR TITLE
Modify systemd unit file reference to align with installation instruction

### PR DESCRIPTION
**Pull request by cgarwood82**
Original file referenced locations not referenced in the installation
instructions. These commits bring that closer to what is prescribed
as well as adding a readme for helping the less experienced.

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [X] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [X] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### Sign off

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
660 York Street, Suite 102,
San Francisco, CA 94110 USA

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Clifford Garwood II cliff@cigii.com

### CODE REVIEW

File: contrib/systemd/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Setup Synapse with Systemd
+This is a setup for managing synapse with the user contributed systemd unit 
+file. It provides a `matrix-synapse` systemd unit file that should be tailored 
+to accomidate your installation in accordinance with the installation
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
to accommodate your installation in accordance with the installation 
```

File: contrib/systemd/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Setup Synapse with Systemd
+This is a setup for managing synapse with the user contributed systemd unit
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
This is a setup for managing synapse with a user contributed systemd unit 
```

File: contrib/systemd/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Setup Synapse with Systemd
+This is a setup for managing synapse with the user contributed systemd unit 
+file. It provides a `matrix-synapse` systemd unit file that should be tailored 
+to accomidate your installation in accordinance with the installation 
+instructions provided in [installation instructions](https://github.com/matrix-org/synapse/blob/master/INSTALL.md).
+
+## Setup
+1. Under the service section, ensure the `User` variable matches which user
+you installed synapse under and wish to run it as. 
+2. Under the service section, ensure the `WorkingDirectory` variablei matches
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
2. Under the service section, ensure the `WorkingDirectory` variable matches
```

File: contrib/systemd/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+where you have installed synapse.
+3. Under the service section, ensure the `ExecStart` variable matches the
+appropriate locations of your installation.
+4. Copy the `mastodon-synapse.service` to `/etc/systemd/system/`
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
4. Copy the `matrix-synapse.service` to `/etc/systemd/system/`
```

:yum:

File: contrib/systemd/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Setup Synapse with Systemd
+This is a setup for managing synapse with the user contributed systemd unit 
+file. It provides a `matrix-synapse` systemd unit file that should be tailored 
+to accomidate your installation in accordinance with the installation 
+instructions provided in [installation instructions](https://github.com/matrix-org/synapse/blob/master/INSTALL.md).
```
Comment (anoadragon453) [inline contrib/systemd/README.md]: You can actually just do this and Github/Gitlab will link it:

```suggestion
instructions provided in [installation instructions](../../INSTALL.md).
```

File: contrib/systemd/matrix-synapse.service [hunk 6]
```diff
@@ -4,8 +4,12 @@
 #    systemctl enable matrix-synapse
 #    systemctl start matrix-synapse
 #
+# This assumes that Synapse has been installed by a user named
+# synapse
+#
 # This assumes that Synapse has been installed in a virtualenv in
-# /opt/synapse/env.
+# the user's home directory
+# /home/synapse/synapse/env.
```
Comment (anoadragon453) [inline contrib/systemd/matrix-synapse.service]:
```suggestion
```

File: contrib/systemd/matrix-synapse.service [hunk 7]
```diff
@@ -4,8 +4,12 @@
 #    systemctl enable matrix-synapse
 #    systemctl start matrix-synapse
 #
+# This assumes that Synapse has been installed by a user named
+# synapse
+#
 # This assumes that Synapse has been installed in a virtualenv in
-# /opt/synapse/env.
+# the user's home directory
```
Comment (anoadragon453) [inline contrib/systemd/matrix-synapse.service]:
```suggestion
# the user's home directory: `/home/synapse/synapse/env`.
```

File: contrib/systemd/matrix-synapse.service [hunk 8]
```diff
@@ -4,8 +4,12 @@
 #    systemctl enable matrix-synapse
 #    systemctl start matrix-synapse
 #
+# This assumes that Synapse has been installed by a user named
+# synapse
```
Comment (anoadragon453) [inline contrib/systemd/matrix-synapse.service]:
```suggestion
# synapse.
```

File: contrib/systemd/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+## Setup
+1. Under the service section, ensure the `User` variable matches which user
+you installed synapse under and wish to run it as. 
+2. Under the service section, ensure the `WorkingDirectory` variablei matches
+where you have installed synapse.
+3. Under the service section, ensure the `ExecStart` variable matches the
+appropriate locations of your installation.
+4. Copy the `mastodon-synapse.service` to `/etc/systemd/system/`
+5. Start the service: `sudo systemctl start matrix-synapse`
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
5. Start Synapse: `sudo systemctl start matrix-synapse`
```

File: contrib/systemd/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+6. Verify the service: `sudo systemctl status matrix-synapse`
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
6. Verify Synapse is running: `sudo systemctl status matrix-synapse`
```

File: contrib/systemd/README.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+7. *optional* Enable the service at boot: `sudo systemctl enable matrix-synapse`
```
Comment (anoadragon453) [inline contrib/systemd/README.md]:
```suggestion
7. *optional* Enable Synapse to start at system boot: `sudo systemctl enable matrix-synapse`
```

**Comment by cgarwood82** [comment]
Thanks @anoadragon453. Thanks for the quality checks! Still says 1 change requested, but it looks like I've approved all of the modifications?

**Comment by anoadragon453** [comment]
This was merged into master, but that would've left it out of CHANGES.md on next release.

I've merged master -> develop, and removed this PR's changelog from master so it doesn't lie around.

File: contrib/systemd/matrix-synapse.service [hunk 12]
```diff
@@ -23,7 +26,7 @@ User=synapse
 Group=nogroup
 
 WorkingDirectory=/opt/synapse
```
Comment (richvdh) [inline contrib/systemd/matrix-synapse.service]: why is `workingdirectory` set to `/opt/synapse` if we're assuming that synapse has been installed in `/home/synapse/synapse` ?

Comment (cgarwood82) [reply contrib/systemd/matrix-synapse.service]: Good catch, I've submitted another PR addressing the issue.

Comment (cgarwood82) [reply contrib/systemd/matrix-synapse.service]: https://github.com/matrix-org/synapse/pull/6490

**Pull request merged**